### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by leifbennett** [projectlombok/lombok#3349]
TL;dr: A problem when compiling with Maven and IDEA (which might not occur with Eclipse) causes a class-level Lombok @Builder annotation to produce incorrect code; a workaround exists.

**Description**
I have a POJO class with several fields, and a Lombok `@Builder` annotation at the class level and also on the all-argument constructor. The fields are declared in one order, and the all-argument constructor declares the corresponding parameters in a different order.

When compiling or running, the behavior indicates that it's attempting to call the constructor with the values in the field order, instead of the parameter order.

**To Reproduce**
For example, this code compiles from Maven or Eclipse, but does not compile in IDEA:

```
package com.ringdna.testutilities;

import lombok.Builder;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;


public class BugReport {

    @Test
    public void demonstrateBug() {

        var bugDemo = BugDemo.builder().id(4L).account(2345L).name("Fred").active(true).build();

        assertThat(bugDemo).isNotNull();
        assertThat(bugDemo.id()).isEqualTo(4L);
    }


    @Builder
    public static class BugDemo {

        private final Long id;
        private final Long account;
        private final String name;
        private final Boolean active;


        @Builder
        public BugDemo(Long id, String name, Long account, Boolean active) {

            this.id = id;
            this.account = account;
            this.name = name;
            this.active = active;
        }


        public Long id() {

            return id;
        }
    }
}

```
When I try to run the unit test, I see this error as it attempts to build:
`incompatible types: java.lang.Long cannot be converted to java.lang.String`

If I instead change the order of the order of the constructor arguments to id, account, name, active (same order as field declaration), it compiles and runs, passing the test as expected.

If I change the order of the constructor arguments to account, id, name, active (swapping account and id from field declaration order), it compiles and runs, but fails the second assertion with the id being 2345; debugging shows that id is 2345 and account is 4. This strongly suggests that the arguments were presented in the order of the fields instead of the order of constructor parameters.

Removing the constructor-level `@Builder` annotation has no effect and the problem remains.

Removing the class-level `@Builder` annotation causes the problem to be solved. I am using this as a workaround.

I see the same behavior using maven to run the test, or using IntelliJ IDEA to run the test. My colleague using Eclipse doesn't seem to see the error.

**Expected Behavior**
The `@Builder` annotation, at class level, should call the constructor correctly.

**Version Info**
Lombok version: 1.18.24
```
% java --version
openjdk 17.0.4.1 2022-08-12 LTS
OpenJDK Runtime Environment Corretto-17.0.4.9.1 (build 17.0.4.1+9-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.4.9.1 (build 17.0.4.1+9-LTS, mixed mode, sharing)
% javac --version
javac 17.0.4.1
```

**Comment by Rawi01** [comment]
As mentioned in the documentation you should add `@Builder` to your custom constructor. Why do you add it to your class and the constructor?

**Comment by leifbennett** [comment]
If I misuse it, I expect an error that says so.

We started with `@Builder` on the class, as first recommended in the documentation I saw. We then ran across a bug report suggesting adding it to the constructor, so we did, and neglected to remove it from the class. So that's how we got to this place that might not be compatible with "the documentation" (but is compatible with [Lombok's `@Builder` documentation](https://projectlombok.org/features/Builder), which states 

> @Builder can be placed on a class, or on a constructor, or on a method. While the "on a class" and "on a constructor" mode are the most common use-case...

).

Might it not be better to work as the user intended? If that's not possible, then it might be better to generate some error relevant to the problem. Silently compiling as if the constructor parameters were in the order of the field parameters is a poor third choice.
